src/groupByKey.ts

```diff
@@ -2,6 +2,23 @@
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+  const arrValues: string[] = [];
+
+  for (const item of items) {
+    const groupKey = item[key] as string;
+
+    if (!arrValues.includes(groupKey)) {
+      arrValues.push(groupKey);
+    }
+  }
+
+  for (const item of arrValues) {
+    const arrFiltered = items.filter((el) => el[key] === item);
+
+    result[item] = arrFiltered;
+  }
```

Comment:
This second loop is unnecessary. You can populate the `result` object directly in the first loop by pushing items into the appropriate group array.